c2corg_ui/static/js/maincontroller.js

```diff
@@ -5,45 +5,25 @@ goog.require('app');
 
 
 /**
+ * @param {angular.Scope} $rootScope The rootScope provider.
  * @param {angularGettext.Catalog} gettextCatalog Gettext catalog.
- * @param {string} langUrlTemplate Language URL template.
  * @constructor
  * @export
  * @ngInject
  */
-app.MainController = function(gettextCatalog, langUrlTemplate) {
+app.MainController = function($rootScope, gettextCatalog) {
 
   /**
-   * @type {angularGettext.Catalog}
-   * @private
+   * TODO: use langController instead?
+   * Put angular-gettext's translate function on the scope to translate
+   * ng-options based dropdown lists. See
+   * https://github.com/rubenv/angular-gettext/issues/58
+   * @param {string} str String to translate.
+   * @return {string} Translated string.
    */
-  this.gettextCatalog_ = gettextCatalog;
-
-  /**
-   * @type {string}
-   * @private
-   */
-  this.langUrlTemplate_ = langUrlTemplate;
-
-  /**
-   * @type {string}
-   * @export
-   */
-  this.lang;
-
-  this.switchLanguage('fr');
-};
-
-
-/**
- * @param {string} lang Language code.
- * @export
- */
-app.MainController.prototype.switchLanguage = function(lang) {
-  this.gettextCatalog_.setCurrentLanguage(lang);
-  this.gettextCatalog_.loadRemote(
-      this.langUrlTemplate_.replace('__lang__', lang));
-  this.lang = lang;
+  $rootScope['translate'] = function(str) {
```

Comment:
The function is indeed duplicated. That would indeed make sense to have it only once.
I have tried to use ``$rootScope.translate()`` in the langDirective but with no luck. Your solution might work, actually I would like to find a way to completely remove the culture-related code from the main controller and have every thing in the langController. Maybe I could simply move ``$rootScope['translate'] = function(str) {`` in the langController instead of the main controller?